src/test/java/org/mskcc/cbio/oncokb/web/rest/UsageAnalysisControllerIT.java

```diff
@@ -0,0 +1,745 @@
+            maxResourceEntry.getKey(),
+            usageSummaryUrl,
+            userIndex,
+            "endpoint"
+          );
+          safeSetNestedValueInFilesObject(
+            maxNoPrivateResourceEntry.getKey(),
+            usageSummaryUrl,
+            userIndex,
+            "noPrivateEndpoint"
+          );
+          safeSetNestedValueInFilesObject(
+            (int) (1000 * ((float) maxResourceEntry.getValue() / totalUsage)) /
+            10f,
+            usageSummaryUrl,
+            userIndex,
+            "maxUsageProportion"
+          );
+          safeSetNestedValueInFilesObject(
+            (int) (
+              1000 * ((float) maxNoPrivateResourceEntry.getValue() / totalUsage)
+            ) /
+            10f,
+            usageSummaryUrl,
+            userIndex,
+            "noPrivateMaxUsageProportion"
+          );
+        }
+
+        safeSetNestedValueInFilesObject(
+          user.getFirstName(),
+          usageUserUrl,
+          "userFirstName"
+        );
+        safeSetNestedValueInFilesObject(
+          user.getLastName(),
+          usageUserUrl,
+          "userLastName"
+        );
+        safeSetNestedValueInFilesObject(
+          user.getEmail(),
+          usageUserUrl,
+          "userEmail"
+        );
+        safeSetNestedValueInFilesObject(
+          userDto.getLicenseType().getName(),
+          usageUserUrl,
+          "licenseType"
+        );
+        safeSetNestedValueInFilesObject(
+          userDto.getJobTitle(),
+          usageUserUrl,
+          "jobTitle"
+        );
+        safeSetNestedValueInFilesObject(
+          userDto.getCompanyName(),
+          usageUserUrl,
+          "company"
+        );
+      }
+    }
+
+    private UserDTO createMockUserDto(User user, CompanyDTO companyDTO) {
+      UserDTO userDto = new UserDTO();
+      userDto.setId(user.getId());
+      userDto.setFirstName(user.getFirstName());
+      userDto.setLastName(user.getLastName());
+      userDto.setEmail(user.getEmail());
+      userDto.setJobTitle(user.getLastName() + " Job Title");
+      userDto.setCompany(companyDTO);
+      userDto.setCompanyName(companyDTO.getName());
+      userDto.setLicenseType(LicenseType.ACADEMIC);
+      return userDto;
+    }
+
+    private CompanyDTO createMockCompany(int companyIndex, String companyName) {
+      CompanyDTO companyDTO = new CompanyDTO();
+      companyDTO.setName(companyName);
+      companyDTO.setId((long) companyIndex);
+      companyDTO.setLicenseType(LicenseType.ACADEMIC);
+      return companyDTO;
+    }
+
+    private User createMockUser(
+      int userIndex,
+      String userName,
+      String companyName
+    ) {
+      User user = new User();
+      user.setId((long) userIndex);
+      user.setEmail(userName + "@" + companyName + ".com");
+      user.setFirstName(String.valueOf(userName.charAt(userName.length() - 1)));
+      user.setLastName(userName.substring(0, userName.length() - 1));
```

Comment:
I plan to make a comment about the logic of parsing username to F/L. But I guess it doesn't matter because we don't really care how user is named?

Reply:
It doesn't matter too much, but the way I have the user name setup is last name ending with the first initial. So `doej` break down into `doe` and `j`.

The only reason why it sort of matter is that if the data is too random then it can be challenging to debug a test so I wanted to have realistic names.